### PR TITLE
Add resource identity_user_membership_v3

### LINKED ISSUE #1142: Add a user to the group
Hi there,

### Terraform Version
```
$ terraform -v
Terraform v0.14.4
+ provider registry.terraform.io/terraform-provider-openstack/openstack v1.35.0
```

### Affected Resources
- openstack_identity_user_v3
- openstack_identity_group_v3

### Terraform Configuration Files
```hcl
data "openstack_identity_user_v3" "user" {
  name = "admin"
}

resource "openstack_identity_group_v3" "group_1" {
  name = "test"
  description = "test"
}
```
Is it possible to add a user `admin` to the group `test` using terraform?

Thanks

### PR DESCRIPTION
**Pull request by nikParasyr**
Add support for new resources indentity_user_membership_v3.
The resources is used to add users to groups.

Close #1142 

Related gophercloud resources:
- https://github.com/gophercloud/gophercloud/blob/master/openstack/identity/v3/users/requests.go#L288
- https://github.com/gophercloud/gophercloud/blob/master/openstack/identity/v3/users/requests.go#L298
- https://github.com/gophercloud/gophercloud/blob/master/openstack/identity/v3/users/requests.go#L313

### CODE REVIEW

File: openstack/resource_openstack_identity_user_membership_v3.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+func resourceIdentityUserMembershipV3Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	identityClient, err := config.IdentityV3Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating OpenStack identity client: %s", err)
+	}
+
+	userID, groupID, err := parseUserMembershipID(d.Id())
+	if err != nil {
+		return CheckDeleted(d, err, "Error parsing user membership ID")
```
Comment (kayrus) [inline openstack/resource_openstack_identity_user_membership_v3.go:70]: Could you please follow the resource name patter in error/log messages? e.g. [here](https://github.com/terraform-provider-openstack/terraform-provider-openstack/blob/d61260067ef0e3b326bb905f7535515b0343c012/openstack/resource_openstack_identity_user_v3.go#L148)

File: website/docs/r/identity_user_membership_v3.html.markdown [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional) The region in which to obtain the V3 Identity client.
+    If omitted, the `region` argument of the provider is used.
+    Changing this creates a new user membership.
```
Comment (kayrus) [inline website/docs/r/identity_user_membership_v3.html.markdown:57]: Please remove two redundant whitespaces in this section

File: website/docs/r/identity_user_membership_v3.html.markdown [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+* `user_id` - (Required) The UUID of user to use. Changing this creates a new user membership.
+
+* `group_id` - (Required) The UUID of group to which the user will be added.
+    Changing this creates a new user membership.
```
Comment (kayrus) [inline website/docs/r/identity_user_membership_v3.html.markdown:62]: And here too.

**Review by kayrus — CHANGES_REQUESTED** [review]
Very good for the first time. I'd request to add an import test for this resource. See [here](https://github.com/terraform-provider-openstack/terraform-provider-openstack/blob/d61260067ef0e3b326bb905f7535515b0343c012/openstack/import_openstack_identity_user_v3_test.go) for example.

File: openstack/resource_openstack_identity_user_membership_v3.go (hunk 1 above)
Comment (nikParasyr) [reply openstack/resource_openstack_identity_user_membership_v3.go]: Thanks :)
Added import test.
Fixed error/log messages

File: website/docs/r/identity_user_membership_v3.html.markdown (hunk 2 above)
Comment (nikParasyr) [reply website/docs/r/identity_user_membership_v3.html.markdown]: fixed

File: website/docs/r/identity_user_membership_v3.html.markdown (hunk 3 above)
Comment (nikParasyr) [reply website/docs/r/identity_user_membership_v3.html.markdown]: fixed

**Comment by kayrus** [comment]
LGTM @ozerovandrei can you review this PR?

**Review by kayrus — APPROVED** [review]

**Review by ozerovandrei — APPROVED** [review]
LGTM

**Pull request merged**